### ISSUE TITLE
feat: add dropdown for more straight-forward search mode switching

### ISSUE THREAD

**Issue by darkcloud784** [homarr-labs/homarr#1795]
### Describe the bug

Both the search icon and the search bar won't let you type anything in them. They only give you the local results. I've even tried creating a search for the overseerr integration but that doesn't show up and again won't let you type anything in the bar.

### Steps to reproduce

deploy current beta using docker compose
import your boards
try to search anything.

### Impact

Makes search non-functional

### Version

1.0.0-beta

### Installation method

Portainer

### Browser

Edge (Chromium)

**Comment by Meierschlumpf** [comment]
Sorry this is not better described yet and we are still on the work of the search functionality (with for example a default search engine) When you want to use a search engine you can type a `!` to see all available engines and then select one. Also you can open a list of all available shortcuts by pressing the `?` while in search.

**Comment by darkcloud784** [comment]
> Sorry this is not better described yet and we are still on the work of the search functionality (with for example a default search engine) When you want to use a search engine you can type a `!` to see all available engines and then select one. Also you can open a list of all available shortcuts by pressing the `?` while in search.

Gotcha, I didn't see that in the docs so I assumed it was a bug. Might I suggest doing away with the ! or ? parameter and have the user select a "default" search. When they want to use one of other search options they can then select them from a drop down menu? Seems this would be more intuitive.

**Comment by Meierschlumpf** [comment]
I guess you are most likely talking from a mobile user perspective. IMO we can add something in that direction for mobile users, but I think for desktop users it's okay that way. What do you think?

**Comment by darkcloud784** [comment]
> I guess you are most likely talking from a mobile user perspective. IMO we can add something in that direction for mobile users, but I think for desktop users it's okay that way. What do you think?


Personally, I think both experiences should follow eachother. So if you did this for mobile, then desktop should follow. I personally I use homarr on both desktop and mobile. I have no issues with the ! or ? parameters but I know my wife does as its not as intuitive for new or less savvy users.

**Comment by Meierschlumpf** [comment]
Makes sense, okay I'll rename this issue to `feat: add dropdown for more straight-forward search mode switching`. Okay?

**Comment by Meierschlumpf** [comment]
Okay you we're actually right. You can't type anything in the home mode of the search right now. I'll fix that combined with the other thing. What do you think of something like this?

![image](https://github.com/user-attachments/assets/c811ba8f-3546-4753-960f-8922c5344d53)

It has a new button in the top right corner to get to the help page (with the combinations and links to documentation and such) and below the default search engine and a possibility to open an other search engine which then opens the search engine selection?
IMO a normal dropdown does not really fit in here
